src/contrasts.jl

```diff
@@ -466,49 +474,79 @@ expressed as a hypothesis matrix, but it is not obvious just from looking at the
 contrasts matrix.  For this reason `HypothesisCoding` is preferred for
 specifying custom contrast coding schemes over `ContrastsCoding`.
 
+Optional arguments `levels` and `labels` give the ordering of mapping between
```

Comment:
"ordering of mapping" isn't very clear to me. Basically, these are the column and row names of the hypothesis matrix, right?

`levels` and `labels` reminds me of the corresponding arguments to `factor` in R, which I always found confusing. Maybe we can find something more explicit than `labels`? Maybe `contrastlabels`?

(BTW, it seems like `NamedArray` inputs could be allowed at some point.)

Reply:
yeah NamedArray would be nice.  but is that a dependency we can take?

`contrastlabels` is a little verbose to my taste so maybe we should get feedback from others on whether `labels` vs. `levels` is clear

Reply:
> yeah NamedArray would be nice. but is that a dependency we can take?

Probably not (not at this point at least), I just wanted to mention that.

Reply:
seems like the dependencies are pretty light, but not nothing (recipesbase, combinatorics, requires).

Reply:
I think the main issue is that the stats ecosystem hasn't really standardized on a package for arrays with dimension names, due to the NamedArrays vs. AxisArrays divide.

Reply:
Though we could imagine defining a set of common functions (e.g. in DataAPI) so that other packages can use names without depending on a particular array package.

Reply:
Ah I see.  I'd like to use something like a NamedArray as teh internal representation of the contrasts matrix, or at least for displaying.  I'd really like to be able to print the contrasts/hypothesis matrices with nice row/column labels but also really don't feel like re-inventing the wheel for that.

Reply:
Yeah it's really annoying that we haven't standardized on something yet.